### PR TITLE
remove bulk publish serially and parallelism metadata

### PR DESCRIPTION
**Pull request by mukundansundar**
Signed-off-by: Mukundan Sundararajan <65565396+mukundansundar@users.noreply.github.com>

# Description
Removes the `bulkPublishSerially` and max concurrency metadata.

## Issue reference

Please reference the issue this PR will close: #5367 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/pubsub/default_bulkpub.go [hunk 1]
```diff
@@ -46,42 +42,17 @@ func NewDefaultBulkPublisher(p contribPubsub.PubSub) *defaultBulkPublisher {
...
-		status := p.bulkPublishSingleEntry(req.PubsubName, req.Topic, entry)
-		if status.Error != nil {
-			err = ErrBulkPublishFailure
-		}
-
-		statuses = append(statuses, status)
-	}
-
-	return contribPubsub.BulkPublishResponse{Statuses: statuses}, err
+	return p.bulkPublishParallel(req)
```
Comment (shubham1172) [inline pkg/runtime/pubsub/default_bulkpub.go:46]: Can we move skip an additional function call and move the parallel implementation here?

File: pkg/runtime/pubsub/default_bulkpub.go [hunk 2]
```diff
@@ -46,42 +42,17 @@ func NewDefaultBulkPublisher(p contribPubsub.PubSub) *defaultBulkPublisher {
...
 
 // bulkPublishParallel publishes messages in parallel. This is faster, but does not guarantee
 // that messages are sent to the broker in the same order as specified in the request.
 func (p *defaultBulkPublisher) bulkPublishParallel(req *contribPubsub.BulkPublishRequest) (contribPubsub.BulkPublishResponse, error) {
 	statuses := make([]contribPubsub.BulkPublishResponseEntry, 0, len(req.Entries))
-	maxConcurrency := utils.GetIntOrDefault(req.Metadata, bulkPublishMaxConcurrencyKey, defaultBulkPublishMaxConcurrency)
 
 	var eg errgroup.Group
-	eg.SetLimit(maxConcurrency)
+	eg.SetLimit(defaultBulkPublishMaxConcurrency)
```
Comment (shubham1172) [inline pkg/runtime/pubsub/default_bulkpub.go:55]: Just evaluating it one last time, is 100 an ok value?

File: pkg/runtime/pubsub/default_bulkpub.go (hunk 1 above)
Comment (shubham1172) [inline pkg/runtime/pubsub/default_bulkpub.go:46]: Also please update the comment to tell it's sent as parallel Publish requests.

**Review by shubham1172 — CHANGES_REQUESTED** [review]

File: pkg/runtime/pubsub/default_bulkpub.go (hunk 1 above)
Comment (mukundansundar) [reply pkg/runtime/pubsub/default_bulkpub.go]: done

File: pkg/runtime/pubsub/default_bulkpub.go (hunk 2 above)
Comment (mukundansundar) [reply pkg/runtime/pubsub/default_bulkpub.go:51]: I believe this to be ok. Unless @dapr/maintainers-dapr @dapr/approvers-dapr have some other thoughts.

File: pkg/runtime/pubsub/default_bulkpub.go (hunk 1 above)
Comment (mukundansundar) [reply pkg/runtime/pubsub/default_bulkpub.go]: done

File: pkg/runtime/pubsub/default_bulkpub.go [hunk 3]
```diff
@@ -46,42 +42,13 @@ func NewDefaultBulkPublisher(p contribPubsub.PubSub) *defaultBulkPublisher {
...
-func (p *defaultBulkPublisher) BulkPublish(_ context.Context, req *contribPubsub.BulkPublishRequest) (contribPubsub.BulkPublishResponse, error) {
-	if utils.IsTruthy(req.Metadata[bulkPublishSeriallyKey]) {
-		return p.bulkPublishSerial(req)
-	} else {
-		return p.bulkPublishParallel(req)
-	}
-}
-
-// bulkPublishSerial publishes messages in a serial order. This is slower, but ensures
+// BulkPublish publishes messages in parallel. This is faster, but does not guarantee
```
Comment (shubham1172) [inline pkg/runtime/pubsub/default_bulkpub.go:45]: Extra line on top.
```suggestion
// BulkPublish publishes a list of messages to a topic as parallel Publish requests.
```

**Review by shubham1172 — APPROVED** [review]

**Review by DeepanshuA — APPROVED** [review]
lgtm

**Review by artursouza — APPROVED** [review]

**Pull request merged**